src/redis_client.rs

```diff
@@ -0,0 +1,107 @@
+use async_trait::async_trait;
+use mobc::{Connection, Pool};
+use mobc_redis::{redis, RedisConnectionManager};
+use std::time::Duration;
+use thiserror::Error;
+use mobc_redis::redis::{RedisError, AsyncCommands};
+use crate::storage::Data;
+
+pub type MobcPool = Pool<RedisConnectionManager>;
+pub type MobcCon = Connection<RedisConnectionManager>;
+
+const CACHE_POOL_MAX_OPEN: u64 = 16;
+const CACHE_POOL_MAX_IDLE: u64 = 8;
+const CACHE_POOL_TIMEOUT_SECONDS: u64 = 1;
+const CACHE_POOL_EXPIRE_SECONDS: u64 = 60;
+
+#[async_trait]
+pub trait FetchCache: Clone + Send + Sync {

```

Comment:
I've always liked the agent noun model that Go uses for its interfaces, i.e. Reader, Writer, Storer, TokenGenerator, etc. What do you think of using CacheFetcher or FetchCacher for this?

Reply:
changed to FetchCacher, I'm not sure "Cacher" is a word but I can roll with it haha